pyomo/util/latex_printer.py

```diff
@@ -0,0 +1,619 @@
+            expressions = [pyomoElement]
+            templatize_fcn = templatize_expression
+
+        if isinstance(pyomoElement, ExpressionBase):
+            objectives = []
+            constraints = []
+            expressions = [pyomoElement]
+            templatize_fcn = templatize_passthrough
+
+        useAlignEnvironment = False
+        isSingle = True
+    else:
+        objectives = [
+            obj
+            for obj in pyomoElement.component_data_objects(
+                pe.Objective, descend_into=True, active=True
+            )
+        ]
+        constraints = [
+            con
+            for con in pyomoElement.component_objects(
+                pe.Constraint, descend_into=True, active=True
+            )
+        ]
+        expressions = []
+        templatize_fcn = templatize_constraint
+
+    # In the case where just a single expression is passed, add this to the constraint list for printing
+    constraints = constraints + expressions
+
+    # Declare a visitor/walker
+    visitor = _LatexVisitor()
+
+    # starts building the output string
+    pstr = ''
+    if useAlignEnvironment:
+        pstr += '\\begin{align} \n'
+        tbSpc = 4
+    else:
+        pstr += '\\begin{equation} \n'
+        if not isSingle:
+            pstr += '    \\begin{aligned} \n'
+            tbSpc = 8
+        else:
+            tbSpc = 4
+
+    # Iterate over the objectives and print
+    for obj in objectives:
+        obj_template, obj_indices = templatize_fcn(obj)
+        if obj.sense == 1:
+            pstr += ' ' * tbSpc + '& \\text{%s} \n' % ('minimize')
+        else:
+            pstr += ' ' * tbSpc + '& \\text{%s} \n' % ('maximize')
+
+        pstr += ' ' * tbSpc + '& & %s ' % (visitor.walk_expression(obj_template))
+        if useAlignEnvironment:
+            pstr += '\\label{obj:' + pyomoElement.name + '_' + obj.name + '} '
+        if not isSingle:
+            pstr += '\\\\ \n'
+        else:
+            pstr += '\n'
+
+    # Iterate over the constraints
+    if len(constraints) > 0:
+        # only print this if printing a full formulation
+        if not isSingle:
+            pstr += ' ' * tbSpc + '& \\text{subject to} \n'
+
+        # first constraint needs different alignment because of the 'subject to'
+        for i in range(0, len(constraints)):
+            if not isSingle:
+                if i == 0:
+                    algn = '& &'
+                else:
+                    algn = '&&&'
+            else:
+                algn = ''
+
+            tail = '\\\\ \n'
+
+            # grab the constraint and templatize
+            con = constraints[i]
+            con_template, indices = templatize_fcn(con)
+
+            # Walk the constraint
+            conLine = (
+                ' ' * tbSpc + algn + ' %s ' % (visitor.walk_expression(con_template))
+            )
+
+            # Multiple constraints are generated using a set
+            if len(indices) > 0:
+                nm = indices[0]._set
```

Comment:
What does `nm` stand for?

Reply:
it stands for set number.  Open to suggestions

Reply:
Deleted the intermediate 'nm' and 'gp' variables